content/articles/2018-10-devtools-2.0.0.Rmarkdown

```diff
@@ -0,0 +1,256 @@
+
+## Conscious uncoupling
+
+This release splits the functionality in devtools into a number of smaller
+packages which are simpler to develop and also easier for other packages to
+depend on. In particular the following packages have been spun off in what we
+are calling the [conscious uncoupling] of devtools.
+
+* [remotes](https://remotes.r-lib.org): Installing packages (i.e.
+  `install_github()`).
+* [pkgbuild](https://github.com/r-lib/pkgbuild): Building binary packages
+  (including checking if build tools are available) (i.e. `build()`).
+* [pkgload](https://github.com/r-lib/pkgload): Simulating package loading (i.e.
+  `load_all()`).
+* [rcmdcheck](https://github.com/r-lib/rcmdcheck): Running R CMD check and
+  reporting the results (i.e. `check()`).
+* [revdepcheck](https://github.com/r-lib/revdepcheck): Running R CMD check on
+  all reverse dependencies, and figuring out what's changed since the last CRAN
+  release (i.e. `revdep_check()`).
+* [sessioninfo](https://github.com/r-lib/sessioninfo): R session info (i.e.
+  `session_info()`).
+* [usethis](https://usethis.r-lib.org): Automating package setup (i.e.
+  `use_test()`).
+
+devtools will remain the main package developers will interact with when
+writing R packages; it will just rely on these other packages internally
+for most of the functionality.
+
+Generally, you should not need to worry about these different packages, because
+devtools installs them all automatically. You will need to care, however, if
+you're filing a bug because reporting it at the correct place will lead to a
+speedier resolution.
+
+Package developers who wish to depend on devtools features should also pay
+attention to which package the functionality is coming from and depend on that
+rather than devtools. In most cases packages should not depend on devtools
+directly. This is similar to the situation with the [tidyverse
+package](https://www.tidyverse.org/articles/2018/06/tidyverse-not-for-packages/).
+
+## New features
+
+The majority of the work for this release was related to bugfixes and
+infrastructure improvements, but there are also some new features you may
+notice.
+
+As always a complete list of all the changes is available in the package
+[Changelog](http://devtools.r-lib.org/news/index.html).
+
+### Upgrade menu
+
+All of the install functions (e.g. `install_github()`) now prompt the user with
+a menu if there are dependencies of the package being installed which are
+outdated. This allows the user to pick which if any they would like to upgrade.
+
+Previous versions of devtools always upgraded these packages automatically by
+default, which sometimes was frustrating when you simply wanted to install one
+package.
+
+When used non-interactively the install functions work like the previous default of
+always upgrading outdated packages.
+
+<p align="center">
+<img src="/images/devtools-2.0.0/upgrade-menu.png" alt="upgrade menu">
```

Comment:
This image is not appearing in the preview but maybe that's just a previewing glitch?

Reply:
Hmm, it works in the local preview at least, I will see if maybe I have it in the wrong place.

Reply:
Oh duh, forgot to commit it 🙈